### DISCUSSION TITLE
Supporting ERC-20 token balances in w3-button

### DISCUSSION THREAD

**Discussion by cpeterson42** [WalletConnect/walletconnect-monorepo#4145] [Developer Support]
I am trying to my ERC-20 token on sepolia and localhost here, but the token balance does not show in my w3-button. I have the token set up in metamask and verified on sepolia etherscan, do you have any suggestions to solve this?

```
createWeb3Modal({
    ethersConfig: defaultConfig({ metadata }),
    chains: [mainnet, sepolia, localhost],
    projectId,
    themeVariables: {
        "--w3m-accent": primaryColor,
        "--w3m-border-radius-master": "4px",
        "--w3m-font-family": "roboto",
    },
    tokens: {
        31337: { address: "0x5FbDB2315678afecb367f032d93F642f64180aa3" },
        11155111: { address: "..." },
    },
})
```

Here is what the button displays (ETH balance on sepolia) 

<img width="368" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/57499623/a4fc5584-79c6-4f5f-84c8-8d59c2db344f">

Any suggestions are appreciated!

**Comment by glitch-txs** [comment]
hi could you open an issue in the Web3Modal repo please?
Thanks! 🙏

**Reply by Ruthjohnson7** [reply]
I can not get to see my keys on the key tab can you please help me
